zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -1,753 +1,49 @@
-        let quotient_range_check = RangeCheckGadget::construct(cb, quotient.expr());
-
-        // Check if the division was done correctly
-        cb.require_equal(
-            "numerator - remainder == quotient ⋅ denominator",
-            numerator - remainder.expr(),
-            quotient.expr() * denominator.expr(),
-        );
-
-        Self {
-            quotient,
-            remainder,
-            denominator,
-            quotient_range_check,
-        }
-    }
-
-    pub(crate) fn quotient(&self) -> Expression<F> {
-        self.quotient.expr()
-    }
-
-    pub(crate) fn remainder(&self) -> Expression<F> {
-        self.remainder.expr()
-    }
-
-    pub(crate) fn assign(
-        &self,
-        region: &mut CachedRegion<'_, '_, F>,
-        offset: usize,
-        numerator: u128,
-    ) -> Result<(u128, u128), Error> {
-        let denominator = self.denominator as u128;
-        let quotient = numerator / denominator;
-        let remainder = numerator % denominator;
-
-        self.quotient
-            .assign(region, offset, Value::known(F::from_u128(quotient)))?;
-        self.remainder
-            .assign(region, offset, Value::known(F::from_u128(remainder)))?;
-
-        self.quotient_range_check
-            .assign(region, offset, F::from_u128(quotient))?;
-
-        Ok((quotient, remainder))
-    }
-}
-
-/// Returns `rhs` when `lhs < rhs`, and returns `lhs` otherwise.
-/// lhs and rhs `< 256**N_BYTES`
-/// `N_BYTES` is required to be `<= MAX_N_BYTES_INTEGER`.
-#[derive(Clone, Debug)]
-pub struct MinMaxGadget<F, const N_BYTES: usize> {
-    lt: LtGadget<F, N_BYTES>,
-    min: Expression<F>,
-    max: Expression<F>,
-}
-
-impl<F: Field, const N_BYTES: usize> MinMaxGadget<F, N_BYTES> {
-    pub(crate) fn construct(
-        cb: &mut ConstraintBuilder<F>,
-        lhs: Expression<F>,
-        rhs: Expression<F>,
-    ) -> Self {
-        let lt = LtGadget::construct(cb, lhs.clone(), rhs.clone());
-        let max = select::expr(lt.expr(), rhs.clone(), lhs.clone());
-        let min = select::expr(lt.expr(), lhs, rhs);
-
-        Self { lt, min, max }
-    }
-
-    pub(crate) fn min(&self) -> Expression<F> {
-        self.min.clone()
-    }
-
-    pub(crate) fn max(&self) -> Expression<F> {
-        self.max.clone()
-    }
-
-    pub(crate) fn assign(
-        &self,
-        region: &mut CachedRegion<'_, '_, F>,
-        offset: usize,
-        lhs: F,
-        rhs: F,
-    ) -> Result<(F, F), Error> {
-        let (lt, _) = self.lt.assign(region, offset, lhs, rhs)?;
-        Ok(if lt.is_zero_vartime() {
-            (rhs, lhs)
-        } else {
-            (lhs, rhs)
-        })
-    }
-}
+use eth_types::Field;
+use halo2_proofs::{circuit::Value, plonk::Expression};
+
+mod abs_word;
+mod addwords;
+mod batched_is_zero;
+mod byte_size;
+mod cmp_words;
+mod comparison;
+mod constant_division;
+mod is_equal;
+mod is_zero;
+mod lt;
+mod lt_word;
+mod min_max;
+mod mod_gadget;
```

Comment:
```suggestion
mod modulo;
```
Naming style: `mod_gadget` vs `modulo`?(anyway `mod` is a keyword of rust